### DISCUSSION TITLE
using the cacheControl mechanism to the runtime js/wasm

### DISCUSSION THREAD

**Discussion by liam-bilich** [jeffreylanters/react-unity-webgl#468] [Question and Answer]
Hi

We need to cache the big files of the runtime:
- build.js
- build.wasm

We are storing the files on CloudFront, with all recommended headers,
but chrome refuses to cache those files (summed as 110mb)

It would be very benficial to be able to utilize the cacheControl
for loading these assets as well, and decide wether to get them from the 
browser cache or invalidate them and download again.

ps. other browsers do cache those files normally (such as safari)
but chrome refuses, as such we want a consistent experiece
for cache control, from the get-go.

Is this possible currently?

**Comment by jeffreylanters** [comment]
Hi! Thanks for your question. Correct me if I'm wrong, but it looks like you're looking for the Custom Cache Control API. I've added support for [this feature](https://react-unity-webgl.dev/docs/api/cache-control) in [last week's release](https://github.com/jeffreylanters/react-unity-webgl/releases/tag/v9.3.0).

**Reply by liam-bilich** [reply]
Hi, yes this is exactly the feature we are referring to.

But from what I understood in the docs it only relates to 
AssetBundle files, and not to the runtime files, or did we get this wrong?

We want the Cache Control API to be utilized for these files as well:

>  loaderUrl: "build/myunityapp.loader.js",
    dataUrl: "build/myunityapp.data",
    frameworkUrl: "build/myunityapp.framework.js",
    codeUrl: "build/myunityapp.wasm",

Thanks

**Reply by jeffreylanters** [reply]
Ah I see. Have you tried enabling Data Caching in your WebGL build settings?

> https://docs.unity3d.com/ScriptReference/PlayerSettings.WebGL-dataCaching.html

**Reply by liam-bilich** [reply]
Not sure I will refer my team to check this and we'll update back 👍

**Reply by liam-bilich** [reply]
It appears that we have this enabled (checked)
but for some reason the browser only caches:
> build.loader.js
> build.framework.js

The files:
> builld.js
build.wasm

Are not cached, and are always re-loaded.
